### PR TITLE
[FLINK-33469] Implement restore tests for Limit node

### PR DESCRIPTION
**Pull request by jnh5y**
## What is the purpose of the change

Implement restore tests for Limit node

## Verifying this change

This change added tests and can be verified as follows:

* Added restore tests for Limit node which verifies the generated compiled plan with the saved compiled plan

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/LimitTestPrograms.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("source_t")
+                                    .addSchema("a INT", "b VARCHAR", "c INT")
+                                    .producedBeforeRestore(DATA)
+                                    .producedAfterRestore(DATA)
+                                    .build())
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink_t")
+                                    .addSchema("a INT", "b VARCHAR", "c BIGINT")
+                                    .addOption("sink-insert-only", "false")
```
Comment (dawidwys) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/LimitTestPrograms.java:50]: This is set by the `RestoreTestBase`

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/LimitTestPrograms.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                Row.of(3, "b", 7),
+                Row.of(5, "c", 9)
+            };
+    static final TableTestProgram LIMIT =
+            TableTestProgram.of("limit", "validates limit node")
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("source_t")
+                                    .addSchema("a INT", "b VARCHAR", "c INT")
+                                    .producedBeforeRestore(DATA)
+                                    .producedAfterRestore(DATA)
```
Comment (dawidwys) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/LimitTestPrograms.java:45]: I'd vote for not having such patterns. See https://github.com/apache/flink/pull/23660#discussion_r1386534891

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/LimitJsonPlanTest.java [hunk 3]
```diff
@@ -1,66 +0,0 @@
...
-                "CREATE TABLE MySink (\n"
-                        + "  a bigint,\n"
-                        + "  b bigint\n"
-                        + ") with (\n"
-                        + "  'connector' = 'values',\n"
-                        + "  'sink-insert-only' = 'false',\n"
-                        + "  'table-sink-class' = 'DEFAULT')";
-        tEnv.executeSql(sinkTableDdl);
-        String sql = "insert into MySink SELECT a, a from MyTable limit 10";
-        util.verifyJsonPlan(sql);
```
Comment (dawidwys) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/LimitJsonPlanTest.java:64]: It uses a checked in plan file, doesn't it? Can we remove it as well?

**Review by dawidwys — COMMENTED** [review]
I think it looks mostly good. 👍

**Comment by jnh5y** [comment]
Responded to comments and moved the new tests to the package `org.apache.flink.table.planner.plan.nodes.exec.stream`.  (Relative to the discussion over on https://github.com/apache/flink/pull/23660.)

**Pull request merged**